CMakeLists.txt

```
@@ -543,12 +543,32 @@ if (ISPC_INCLUDE_UTILS)
     endif()
 endif()
 
-if (ISPC_INCLUDE_EXAMPLES AND NOT ISPC_PREPARE_PACKAGE)
-    add_subdirectory(examples)
+if ( ISPC_INCLUDE_EXAMPLES AND NOT ISPC_PREPARE_PACKAGE)
+    include(ExternalProject)
+    ExternalProject_Add(ispc_examples
+        PREFIX examples
+        DEPENDS ispc
+        SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/examples"
+        CMAKE_CACHE_ARGS
+            -DISPC_BUILD:BOOL=TRUE
+            -DCMAKE_ISPC_COMPILER:PATH=$<TARGET_FILE:ispc>
+            -DISPC_EXECUTABLE:PATH=$<TARGET_FILE:ispc>
```

```suggestion
            -DISPC_EXECUTABLE:PATH=$<TARGET_FILE:ispc>
            -DCMAKE_C_COMPILER:PATH=${CMAKE_C_COMPILER}
            -DCMAKE_CXX_COMPILER:PATH=${CMAKE_CXX_COMPILER}
```

By default we should use clang used for ISPC build.